src/game/configuration/settings.ts

```diff
@@ -9,7 +9,19 @@ export const collisionPointOnBallClass = "bba-collision-point-on-ball"
 export type StartOptions = {
   withScoreboard: boolean
   initialBallSpeed: "low" | "middle" | "high" | "superHigh"
-  sound: boolean
+  sound: boolean,
+  visualizeBlocks: boolean,
+  controlMode: "normal" | "mouse"
+}
```

Comment:
[imo] How about a structure like this?👀 I think it might be more robust.
```suggestion
  sound: boolean
} & {
  visualizeBlocks: false,
  controlMode: "normal"
} | {
  visualizeBlocks: true,
  controlMode: "mouse"
} 
```

Reply:
🤔 I'm undecided about this.

I found two bugs using the `controlMode: "mouse"` feature. I actually couldn't  find one of the bugs actually because the borders drawn by `visualizeBlocks` prevent me from reproducing it (that's why I left the TODO "More customizable debugging mode: make controlMode and visualizeBlocks separately configurable on popup.html.").

But now, I know how to fix it and the change to fix it would perhaps remove the root problem of `visualizeBlocks`. So separating them could just make the code more complex.

I beg your opinion!

Reply:
Rethinking this problem, now I want to keep as is.

- As my need for separating the debug options has already been satisfied by (manually editing the code), we would rarely need to switch them independently. So keep popup.tsx simple by not adding new `<input>`s for the time being.
- But IMO `visualizeBlocks` and `controlMode` are essentially different. They should be configured independently (even by temporarily updating the code). Your suggestion makes it a little harder to do that.